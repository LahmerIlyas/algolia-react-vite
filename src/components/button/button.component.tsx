/* eslint-disable react/prop-types */
interface ButtonProps {
  variant: 'primary' | 'secondary'
}

export const Button: React.FC<ButtonProps> = (props) => {
  return (
    <button className="px-4 py-2 rounded-2 border-2 font-medium bg-green-200">
      {props.children}
    </button>
  )
}
