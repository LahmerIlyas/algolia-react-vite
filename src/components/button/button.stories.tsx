import React from 'react'
import type { ComponentMeta, ComponentStory } from '@storybook/react'

import { Button } from './button.component'

export default {
  title: 'Example/Button',
  component: Button,
} as ComponentMeta<typeof Button >

const Template: ComponentStory<typeof Button> = () => (
  <React.Fragment>
    <Button variant="primary">
      Click me
    </Button>
  </React.Fragment>
)

export const ExampleStory = Template.bind({})
ExampleStory.args = {
}
