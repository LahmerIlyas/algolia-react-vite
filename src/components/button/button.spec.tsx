import { describe, expect, it } from 'vitest'
import renderer from 'react-test-renderer'
import { Button } from './button.component'

describe('Given button element', () => {
  it('should renders primary variant correctly', () => {
    const tree = renderer.create(<Button variant="primary" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('should renders secondary variant correctly', () => {
    const tree = renderer.create(<Button variant="secondary" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
