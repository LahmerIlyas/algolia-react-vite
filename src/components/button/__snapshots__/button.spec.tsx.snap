// Vitest Snapshot v1

exports[`Given button element > should renders primary variant correctly 1`] = `
<button
  className="px-4 py-2 rounded-2 border-2 font-medium bg-green-200"
/>
`;

exports[`Given button element > should renders secondary variant correctly 1`] = `
<button
  className="px-4 py-2 rounded-2 border-2 font-medium bg-green-200"
/>
`;
